src/Symfony/Component/PropertyAccess/Mapping/ClassMetadata.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Mapping;
+
+/**
+ * @author Luis Ramón López <lrlopez@gmail.com>
+ */
+class ClassMetadata
+{
+    /**
+     * @var string
+     *
+     * @internal This property is public in order to reduce the size of the
+     *           class' serialized representation. Do not access it. Use
+     *           {@link getName()} instead.
+     */
+    public $name;
+
+    /**
+     * @var PropertyMetadata[]
+     *
+     * @internal This property is public in order to reduce the size of the
+     *           class' serialized representation. Do not access it. Use
+     *           {@link getPropertyMetadataCollection()} instead.
+     */
+    public $propertyMetadataCollection = array();
+
+    /**
+     * @var \ReflectionClass
+     */
+    private $reflClass;
+
+    /**
+     * Constructs a metadata for the given class.
+     *
+     * @param string $class
+     */
+    public function __construct($class)
+    {
+        $this->name = $class;
+    }
+
+    /**
+     * Returns the name of the backing PHP class.
+     *
+     * @return string the name of the backing class
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Adds an {@link AttributeMetadataInterface}.
+     *
+     * @param PropertyMetadata $propertyMetadata
+     */
+    public function addPropertyMetadata(PropertyMetadata $propertyMetadata)
```

Comment:
Do you think we can make this value object immutable? We do that for new VO (PSR-7 style, see the PropertyInfo component for instance).

Reply:
It could be done. But I'm curious about what would be the benefits. This class is used internally only, it's not exposed outside the component. Some parts of this code are borrowed from the `Serializer` component but I infer that the current implementation intent is to allow fast & easy merging of metadata from different sources and classes in the hierarchy.